planemo/shed.py

```diff
@@ -193,23 +200,132 @@ def _find_repository_id(ctx, tsi, name, repo_config, **kwds):
+    # If repositories aren't defined, just define a single
+    # one based on calculated name and including everything
+    # by default.
+    if repos is None:
+        repos = {
+            config["name"]: {
+                "include": default_include
+            }
+        }
+    if suite_config:
+        _build_suite_repo(config, repos, suite_config)
+    config["repositories"] = repos
+
+
+def _build_auto_tool_repos(path, config, auto_tool_repos):
+    default_include = config.get("include", ["**"])
+    tool_els = list(load_tool_elements_from_path(path, recursive=True))
+    paths = list(map(lambda pair: pair[0], tool_els))
+    excludes = _shed_config_excludes(config)
+
+    def _build_repository(tool_path, tool_el):
+        tool_id = tool_el.getroot().get("id")
+        tool_name = tool_el.getroot().get("name")
+        template_vars = dict(
+            tool_id=tool_id,
+            tool_name=tool_name,
+        )
+        other_paths = paths[:]
+        other_paths.remove(tool_path)
+        tool_excludes = excludes + list(other_paths)
+        repo_dict = {
+            "include": default_include,
+            "exclude": tool_excludes,
+        }
+        for key in ["name", "description", "long_description"]:
+            template_key = "%s_template" % key
+            template = auto_tool_repos.get(template_key, None)
+            if template:
+                value = templates.render(template, **template_vars)
+                repo_dict[key] = value
+        return repo_dict
+
+    repos = {}
+    for tool_path, tool_el in tool_els:
+        repository_config = _build_repository(tool_path, tool_el)
+        repository_name = repository_config["name"]
+        repos[repository_name] = repository_config
+    return repos
+
+
+def _build_suite_repo(config, repos, suite_config):
+    if not isinstance(suite_config, dict):
+        suite_config = {}
+
+    name = suite_config.get("name", None)
+    if name is None:
+        raise Exception("suite_configitories required name key.")
+    description = suite_config.get("description", "")
+    long_description = suite_config.get("long_description", None)
+    owner = config["owner"]
+
+    repo_pairs = map(lambda name: (owner, name), repos.keys())
+    extra_repos = suite_config.get("include_repositories", {})
+    extra_pairs = map(lambda item: (item["owner"], item["name"]), extra_repos)
+
+    repository_dependencies = RepositoryDependencies()
+    repository_dependencies.description = description
+    repository_dependencies.repo_pairs = list(repo_pairs) + list(extra_pairs)
+
+    repo = {
+        "_files": {
+            REPO_DEPENDENCIES_CONFIG_NAME: str(repository_dependencies)
+        },
+        "include": [],
+        "name": name,
+        "description": description,
+    }
+    if long_description:
+        repo["long_description"] = long_description
+    repos[name] = repo
+
+
+def find_repository(tsi, owner, name):
```

Comment:
Oh, I meant the shed.py file in planemo where a bunch of TS interactivity
was stuck

man. 27. apr. 2015, 08.32 skrev John Chilton <notifications@github.com>:

> In planemo/shed.py
> <https://github.com/galaxyproject/planemo/pull/143#discussion_r29145707>:
>
> > +    repository_dependencies.repo_pairs = list(repo_pairs) + list(extra_pairs)
> > +
> > +    repo = {
> > +        "_files": {
> > +            REPO_DEPENDENCIES_CONFIG_NAME: str(repository_dependencies)
> > +        },
> > +        "include": [],
> > +        "name": name,
> > +        "description": description,
> > +    }
> > +    if long_description:
> > +        repo["long_description"] = long_description
> > +    repos[name] = repo
> > +
> > +
> > +def find_repository(tsi, owner, name):
>
> What is shed util?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/galaxyproject/planemo/pull/143/files#r29145707>.
>